packages/dds/sequence/src/test/intervalCollection.fuzz.spec.ts

```diff
@@ -0,0 +1,390 @@
+import { SharedString } from "../sharedString";
+import { IntervalCollection, IntervalType, SequenceInterval } from "../intervalCollection";
+import { SharedStringFactory } from "../sequenceFactory";
+
+interface FuzzTestState extends BaseFuzzTestState {
+    containerRuntimeFactory: MockContainerRuntimeFactory;
+    sharedStrings: SharedString[];
+}
+
+interface ClientSpec {
+    stringId: string;
+}
+
+interface RangeSpec {
+    start: number;
+    end: number;
+}
+
+interface IntervalCollectionSpec {
+    collectionName: string;
+}
+
+interface AddText extends ClientSpec {
+    type: "addText",
+    index: number;
+    content: string;
+}
+
+interface RemoveRange extends ClientSpec, RangeSpec {
+    type: "removeRange"
+}
+
+// For non-interval collection fuzzing, annotating text would also be useful.
+
+interface AddInterval extends ClientSpec, IntervalCollectionSpec, RangeSpec {
+    type: "addInterval"
+}
+
+interface ChangeInterval extends ClientSpec, IntervalCollectionSpec, RangeSpec {
+    type: "changeInterval";
+    id: string;
+}
+
+interface DeleteInterval extends ClientSpec, IntervalCollectionSpec {
+    type: "deleteInterval",
+    id: string;
+}
+
+interface Synchronize {
+    type: "synchronize";
+}
+
+type IntervalOperation = AddInterval | ChangeInterval | DeleteInterval;
+
+type TextOperation = AddText | RemoveRange;
+
+type ClientOperation = IntervalOperation | TextOperation;
+
+type Operation = ClientOperation | Synchronize;
+
+// Note: none of these options are currently exercised, since the fuzz test fails with pretty much
+// any configuration due to known bugs. Once shared interval collections are in a better state these
+// should be revisited.
+interface OperationGenerationConfig {
+    /**
+     * Maximum length of the SharedString (locally) before no further AddText operations are generated.
+     * Note due to concurency, during test execution the actual length of the string may exceed this.
+     */
+    maxStringLength?: number;
+    /**
+     * Maximum number of intervals (locally) before no further AddInterval operations are generated.
+     * Note due to concurency, during test execution the actual number of intervals may exceed this.
+     */
+    maxIntervals?: number;
+    maxInsertLength?: number;
+    intervalCollectionNamePool?: string[];
+    validateInterval?: number;
+}
+
+const defaultOptions: Required<OperationGenerationConfig> = {
+    maxStringLength: 1000,
+    maxIntervals: 100,
+    maxInsertLength: 10,
+    intervalCollectionNamePool: ["comments"],
+    validateInterval: 100,
+};
+
+// A few places in the fuzz testing code need to inspect existing collections on a SharedString
+// to determine where to insert/modify/delete an interval from. There's some logic to scope all
+// provided labels with "intervalCollections/", which is what the iterator returns. This works
+// around that.
+function* getUnscopedLabels(string: SharedString): Iterable<string> {
```

Comment:
this seems like it should just be done on `getIntervalCollectionLabels`, but I'm going to push that change out to a separate PR. Probably nobody depends on the behavior, but there is a unit test for it so.... best to have that discussion separately 😛

Reply:
after looking more closely, the create event gets raised by map kernel I think, so yielding these instead seems more reasonable. It's still a pretty awkward scheme, but doesn't seem worth fixing since we hope to rip out mapKernel anyway.